lib/mix/lib/mix/dep.ex

```diff
@@ -132,6 +132,17 @@ defmodule Mix.Dep do
     seen = populate_seen(MapSet.new(), [app])
     children = get_deps(deps, tl(Enum.uniq(get_children(deps, seen, [app]))))
 
+    if bottom.project()[:apps_path] do
+      umbrella_apps =
+        deps
+        |> Enum.filter(& &1.opts[:from_umbrella])
+        |> Enum.map(& &1.app)
+
+      app in umbrella_apps ||
+        raise "cannot load #{inspect(app)} app because the defined app name in mix.exs and the app directory name are different. " <>
```

Comment:
Note inside Mix we use `Mix.raise` and the error messages should start in upcase.